src/dotnet/commands/dotnet-compile/CompilerUtil.cs

```diff
@@ -26,12 +26,29 @@ public static string ResolveCompilerName(ProjectContext context)
 
             return compilerName;
         }
-        
+
+        private static readonly KeyValuePair<string, string>[] s_compilerNameToLanguageId =
+        {
+            new KeyValuePair<string, string>("csc", "cs"),
+            new KeyValuePair<string, string>("vbc", "vb"),
+            new KeyValuePair<string, string>("fsc", "fs")
+        };
```

Comment:
No, this is just the default. You can override it manually if you have a new language.